### ISSUE TITLE
Invalid union when intervals share a bound, one open and one closed

### ISSUE THREAD

**Issue by AlexandreDecan** [AlexandreDecan/python-intervals#12]
Unioning `AtomicInterval` instances (and by extension `Interval` instances) can  lead to an invalid result if the two intervals share a (lower or upper) bound, where one is open and the other one is closed. 

```python
>>> I.closed(0, 2) | I.open(0, 2)
I.open(0, 2)
```
... where the expected result is `I.closed(0, 2)`.

**Comment by AlexandreDecan** [comment]
This issue is fixed in the master branch, and the fix will be part of 1.9.0 update or, if I cannot manage to release this version soon, will be hotfixed in 1.8.1.

**Comment by AlexandreDecan** [comment]
Not yet fixed, for example:

```python
>>> I.closed(0, 2) | I.open(1, 2) == I.openclosed(0, 2)
False
```

Should be `True`.

**Comment by AlexandreDecan** [comment]
Fixed in 1.9.0, now available on PyPI.
